src/app/Context/CryptoContext.jsx

```diff
@@ -4,63 +4,71 @@ import { createContext, useState, useContext } from "react";
 export const CryptoContext = createContext();
 
 export function useCrypto() {
-  const value = useContext(CryptoContext);
-  return value;
+   const value = useContext(CryptoContext);
+   return value;
 }
 
 export const CryptoProvider = ({ children }) => {
-  const [marketData, setMarketData] = useState([]);
-  const [userAssetData, setUserAssetData] = useState([]);
-  const [login, setLogin] = useState();
-  const [currency, setCurrency] = useState("usd");
-  const [listOfUsers, setListOfUsers] = useState([]);
-  const [globalData, setGlobalData] = useState();
-  const [darkMode, setDarkMode] = useState(true);
+   const [marketData, setMarketData] = useState([]);
+   const [userAssetData, setUserAssetData] = useState([]);
+   const [login, setLogin] = useState();
+   const [currency, setCurrency] = useState("usd");
+   const [listOfUsers, setListOfUsers] = useState([]);
+   const [globalData, setGlobalData] = useState();
+   const [darkMode, setDarkMode] = useState(true);
 
-  const saveUserList = () => {
-    localStorage.setItem("users", JSON.stringify(listOfUsers));
-  };
+   const saveUserList = () => {
+      localStorage.setItem("users", JSON.stringify(listOfUsers));
+   };
 
-  const loadUserList = () => {
-    const users = JSON.parse(localStorage.getItem("users"));
-    if (users) {
-      setListOfUsers(users);
-    } else {
-      setListOfUsers([]);
-    }
-  };
+   const loadUserList = () => {
+      const users = JSON.parse(localStorage.getItem("users"));
+      if (users) {
+         setListOfUsers(users);
+      } else {
+         setListOfUsers([]);
+      }
+   };
 
-  const saveUserData = (login) => {
-    const loadedUser = listOfUsers.find((user) => user.id === login.id);
-    const index = listOfUsers.findIndex((user) => user.id === login.id);
-    loadedUser.portfolio = login.portfolio;
-    listOfUsers[index] = loadedUser;
-    saveUserList();
-  };
+   const saveDarkMode = (login) => {
+      const loadedUser = listOfUsers.find((user) => user.id === login.id);
+      const index = listOfUsers.findIndex((user) => user.id === login.id);
+      loadedUser.dark_mode = login.dark_mode;
+      listOfUsers[index] = loadedUser;
+      saveUserList();
+   };
 
-  return (
-    <CryptoContext.Provider
-      value={{
-        marketData,
-        setMarketData,
-        userAssetData,
-        setUserAssetData,
-        currency,
-        setCurrency,
-        listOfUsers,
-        setListOfUsers,
-        saveUserList,
-        login,
-        setLogin,
-        loadUserList,
-        saveUserData,
-        globalData,
-        setGlobalData,
-        darkMode,
-        setDarkMode,
-      }}
-    >
-      {children}
-    </CryptoContext.Provider>
-  );
+   const saveUserData = (login) => {
+      const loadedUser = listOfUsers.find((user) => user.id === login.id);
+      const index = listOfUsers.findIndex((user) => user.id === login.id);
+      loadedUser.portfolio = login.portfolio;
+      listOfUsers[index] = loadedUser;
```

Comment:
this is mutating state directly, instead you could have use a .map and then update the state with the setListOfUsers